src/transformations.jl

```diff
@@ -0,0 +1,32 @@
+abstract type AbstractCoordinateTransformation{N, CT<:AbstractCoordinateSystem} end
+abstract type AbstractAffineMap{N,CT} <: AbstractCoordinateTransformation{N,CT} end
+
+###
+# Translation
+###
+struct Translation{N, CT<:AbstractCoordinateSystem, V} <: AbstractAffineMap{N,CT}
+    cs::CT
+    translation::V
+    function Translation{N, CT, V}(cs::CT, translation::V) where {N,CT<:AbstractCoordinateSystem,V}
+        if N != length(translation)
+            throw(ArgumentError("Coordinate system $(CT) dimension $(ndims(cs)) should be equal to translation vector length $(length(translation))."))
+        end
+        new{N,CT,V}(cs, translation)
+    end
+end
+@inline function Translation(cs::CT, x) where {CT<:AbstractCoordinateSystem}
+    x = SVector(x)
+    return Translation{ndims(CT),CT,typeof(x)}(cs, x)
+end
+@inline Translation(x) = Translation(coordinate_system(x), x)
+@inline Translation(xs::Real...) = Translation(SVector(xs))
+
+function (trans::Translation)(x::AbstractPoint)
+    with_coordinate_system(trans.cs, x) do p
+        SVector(p) + trans.translation
+    end
+end
```

Comment:
I'm not sure here if we want the output point live in the coordinate system of `trans` or live in the coordinate system of input `x`. The current implementation chooses the latter one but obviously, if we're having mixed coordinate systems during computation there will be performance overhead.